### DISCUSSION TITLE
Implementing your own Peer to Peer Network for sharing files in a private network

### DISCUSSION THREAD

**Discussion by moksh98** [arvidn/libtorrent#7306] [Q&A]
Hi,

I am relatively new to the libtorrent library and don't know C++.

I have a general idea of how p2p network functions and this is what I am trying to achieve.

I have a large file that I would like to share across devices in my home network. I want to create my own tracker, allow peers to connect to it, and start downloading and uploading files into the network.

There needs to be one device that has the whole file and is seeding this file. Peers need to register with the tracker and start downloading this file.

I would like to implement this in python and any help would be appreciated.

Thanks in advance.

**Comment by arvidn** [comment]
For a local network, you could also rely on "local service discovery" (lsd). This is a feature where you can find peers via multi-cast on a local network. It might simplify things as you wouldn't need a tracker. However, it wouldn't work on networks that disable multicast or block it between segments.

You would still need to distribute the .torrent file (or magnet link) to the peers though.
